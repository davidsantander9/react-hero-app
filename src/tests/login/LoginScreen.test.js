import React from 'react';
import { mount } from 'enzyme';
import { LoginScreen } from "../../components/login/LoginScreen";
import { AuthContext } from '../../auth/AuthContext';
import { types } from '../../types/types';

describe('test on <LoginScreen/>', () => {

    const contextValue = {
        dispatch: jest.fn(),
        user:{
            logged: false
        }
    }

    
    const history = {
        replace: jest.fn(),
    }

    const wrapper = mount( 
        <AuthContext.Provider  value = { contextValue } >
            <LoginScreen history={ history }/>
        </AuthContext.Provider>
    );


    test('should show it correctly', () => {
       expect( wrapper ).toMatchSnapshot();
    });
    

    test('should did dispath and navegated', () => {
        
        wrapper.find('button').prop('onClick')();
        expect( contextValue.dispatch ).toHaveBeenCalledWith({
            type: types.login,
            payload:{
                name: 'David',
            }
        });

        expect( history.replace ).toHaveBeenCalled();

    });
    

});
