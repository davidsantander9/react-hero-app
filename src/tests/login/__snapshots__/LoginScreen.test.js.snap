// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on <LoginScreen/> should show it correctly 1`] = `
<div
  className="container mt-5"
>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
